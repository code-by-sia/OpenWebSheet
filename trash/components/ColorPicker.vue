<script lang="ts">
import { Component, Prop } from 'vue-property-decorator';
import Vue from 'vue';

@Component({name: 'ColorPicker'})
export default class ColorPicker extends Vue {
  @Prop({default: '#ffffff'})
  public value!: string;

  @Prop()
  public label?: string;

  get color() {
    if (!this.value) {
      return '#ffffff';
    }
    return this.value;
  }

  set color(newValue) {
    this.$emit('input', newValue);
  }
}
</script>
<template>
    <div class="color-picker" data-qa="color-picker">
        <label data-qa="label">
            <slot>
                <span class="prefix" v-if="!!$slots['prefix']">
                    <slot name="prefix"/>
                </span>
                {{label}}
                <slot name="postfix" class="postfix"/>
            </slot>
        </label>
        <input type="color" v-model="color" data-qa="input-el"/>
    </div>
</template>
<style lang="scss" scoped>
    .color-picker {
        display: flex;

        & label {
            display: flex;
            align-items: center;
            margin-right: 5px;
        }

        & .prefix {
            margin-right: 5px;
        }

        & .postfix {
            margin-left: 5px;
        }

        & input[type=color] {
            border: none;
            padding: 0;
            width: 20px;
        }
    }
</style>
