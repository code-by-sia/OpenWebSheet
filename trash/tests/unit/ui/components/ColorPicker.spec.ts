import { shallowMount } from '@vue/test-utils';
import ColorPicker from '@/components/ColorPicker.vue';

describe('ColorPicker', () => {
  it('is a vue instance', () => {
    const wrapper = shallowMount(ColorPicker);
    expect(wrapper.isVueInstance()).toBe(true);
  });

  it('should use white as default value', () => {
    const wrapper = shallowMount(ColorPicker);
    expect(wrapper.props('value')).toBe('#ffffff');
    expect(wrapper.props('label')).toBeUndefined();
  });

  it('should renders label', () => {
    const wrapper = shallowMount(ColorPicker, {
      propsData: {
        label: 'LABEL',
      },
    });
    expect(wrapper.find('[data-qa=label]').text()).toBe('LABEL');
  });

  it('should renders default slot when it is exits', () => {
    const wrapper = shallowMount(ColorPicker, {
      slots: {
        default: '[SLUG]',
      },
    });
    expect(wrapper.find('[data-qa=label]').text()).toBe('[SLUG]');
  });

  it('should renders pre/post slots when they are exits', () => {
    const wrapper = shallowMount(ColorPicker, {
      slots: {
        prefix: '[PRE]',
        postfix: '[POST]',
      },
      propsData: {
        label: 'LABEL',
      },
    });
    expect(wrapper.find('[data-qa=label]').text()).toMatchSnapshot();
  });

  it('should use the prop value for input value', () => {
    const wrapper = shallowMount(ColorPicker, {
      propsData: {
        value: '#00ff00',
      },
    });
    expect((wrapper.find('[data-qa=input-el]').element as HTMLInputElement).value).toBe('#00ff00');
  });


  it('should use WHITE when the value is null', () => {
    const wrapper = shallowMount(ColorPicker, {
      propsData: {
        value: null,
      },
    });
    expect((wrapper.find('[data-qa=input-el]').element as HTMLInputElement).value).toBe('#ffffff');
  });


  it('should emit change when the color changes', () => {
    const wrapper = shallowMount(ColorPicker);
    wrapper.find('[data-qa=input-el]').setValue('#ff0000');
    expect(wrapper.emitted('input')).toEqual([['#ff0000']]);
  });

});
