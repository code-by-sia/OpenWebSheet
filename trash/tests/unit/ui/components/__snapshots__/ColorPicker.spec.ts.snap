// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker should renders pre/post slots when they are exits 1`] = `
"[PRE]
            LABEL
            [POST]"
`;
